baseplate/lint/db_query_string_format_plugin.py

```diff
@@ -0,0 +1,89 @@
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class NoDbQueryStringFormatChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "no-database-query-string-format"
+    priority = -1
+    msgs = {
+        "W9000": (
+            "Python string formatting found in Cassandra database query. Database queries should be using Cassandra driver parameter substitution.",
+            "database-query-string-format",
+            "This allows CQL/SQL injection.",
+        )
+    }
+
+    def __init__(self, linter: PyLinter = None):
+        super().__init__(linter)
+        self.string_sub_queries: set = set()
+        self.query_verbs = {"select", "update", "delete", "insertinto", "truncate"}
+
+    def check_string_is_query(self, string: str) -> bool:
+        query = string.split(" ")
+        if query[0].lower() in self.query_verbs and ("{}" or "%s" in query):
```

Comment:
Note, I don't think this will catch f-strings or the case where you name the format values, both of which are `"{some_name_or_code}"`.